src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -672,13 +694,22 @@ bool AdaptDispatch::EraseCharacters(const size_t numChars)
     {
         const COORD startPosition = csbiex.dwCursorPosition;
 
-        const size_t remainingSpaces = csbiex.dwSize.X - startPosition.X;
-        const auto actualRemaining = (remainingSpaces < 0) ? 0 : remainingSpaces;
-        // erase at max the number of characters remaining in the line from the current position.
-        const auto eraseLength = (numChars <= actualRemaining) ? numChars : actualRemaining;
+        SHORT result{ 0 };
+        success = SUCCEEDED(ShortSub(csbiex.dwSize.X, startPosition.X, &result));
+        if (success)
+        {
+            size_t remainingSpaces{ 0 };
+            success = SUCCEEDED(ShortToSizeT(result, &remainingSpaces));
```

Comment:
This feels very wrong. It's something like 9 lines of boilerplate to "safely" subtract two numbers that should never overflow in the first place. If we're just trying to get rid of that "arithmetic overflow" warning (which I suspect is actually a compiler bug), then simply changing the `remainingSpaces` from `size_t` to `auto` seems to do the trick. That would also fix the one real issue with the code, which is that the `actualRemaining` calculation is pointless if `remainingSpaces` is an unsigned type.

Reply:
Yeah. We need to do a concerted effort on this with #4013 but I'm going to try my best to make it not-much-worse with getting audit on.